detection_rules/rule.py

```diff
@@ -909,6 +910,22 @@ def validates_esql_data(self, data, **kwargs):
         if data.get('index'):
             raise ValidationError("Index is not a valid field for ES|QL rule type.")
 
+        # Convert the query string to lowercase to handle case insensitivity
+        query_lower = data['query'].lower()
+
+        # Metadata fields that are required for non-aggregate queries
+        metadata_pattern = r'from\s+\S+\s+metadata\s+_id,\s*_version,\s*_index'
+        # Allow any characters (including newlines and complex expressions) between stats and by
+        stats_pattern = r'\bstats\b.*?\bby\b'
+
+        # Ensure that non-aggregate queries have metadata
+        if not re.search(stats_pattern, query_lower, re.DOTALL) and not re.search(metadata_pattern, query_lower):
+            raise ValidationError(
+                f"Rule: {data['name']} contains a non-aggregate query without"
+                f"metadata fields '_id', '_version', and '_index' ->"
```

Comment:
a space missing between `without` and `metadata`

Reply:
Thanks! Updated.